setup.js

```diff
@@ -1,52 +1,96 @@
-//
-// Data
-//
-
-function activity(name, location, hours, isIndoor, intensity, cost) {
-  this.name = name;
-  this.location = location;
-  this.hours = hours;
-  this.isIndoor = isIndoor;
-
-  //intensity is "low" (sitting), "medium" (walking), "high" (very active)
-  this.intensity = intensity;
-
-  this.cost = cost;
-}
-
-var listOfActivities = [
-  new activity("Oregon Zoo", "Washington Park", "9am-5pm", false, "medium", "$15"),
-  new activity("Voodoo Donuts"),
-  new activity("Children's Museum"),
-  new activity("Art Museum"),
-  new activity("Evergreen Aviation Museum"),
-  new activity("Tillamook Creamery"),
-  new activity("Newport Aquarium"),
-  new activity("Pioneer Place"),
-  new activity("Crystal Ballroom"),
-  new activity("Mississippi Studio"),
-  new activity("MODA Center"),
-  new activity("Multnomah Falls"),
-  new activity("Cannon Beach"),
-  new activity("Washington Park"),
-  new activity("Rose Garden"),
-  new activity("Bonneville Dam"),
-  new activity("Newport Aquarium"),
-  new activity("Oaks Amusement Park"),
-  new activity("Saturday Market"),
-  new activity("The Grotto"),
-  new activity("Mt. Hood"),
-  new activity("Portland Kayak Tours"),
-  new activity("BrewCycle"),
-];
-
-var tags = {
-  kidFriendly: ["Oregon Zoo", "Children's Museum", "Art Museum",
-                "Evergreen Aviation Museum", "Tillamook Creamery",
-                "Newport Aquarium", "Pioneer Place",
-                "Multnomah Falls", "Cannon Beach", "Rose Garden",
-                "Oaks Amusement Park", "The Grotto", "Saturday Market",
-                "Mt. Hood", "Bonneville Dam", "Washington Park",
-                "International Rose Garden"],
-  adult: ["BrewCycle"]
-};
+var App = {};
+
+// Use the App.functions below to access the data
+// For example, call the App.allActivityNames() function if you need the names of each activity to use for your functions
+// This way, you don't need to parse the object/arrays for this information
+
+(function() {
+  // Returns a list of all activities in alphabetical order
+  App.allActivities = function() {
+    return listOfActivities.sort(function(a, b) {
+      return a.name.localeCompare(b.name);
+    });
+  };
+
+  // Returns all the activity names in alphabetical order
+  App.allActivityNames = function() {
+    return listOfActivities.map(function(activity) {
+      return activity.name;
+    }).sort();
+  };
+
+  // Return all activities that have the given tag in alphabetical order
+  // @param tag: a string with the tag name, e.g. "kidFriendly" or "adult"
+  App.activitiesForTag = function(tag) {
+    return tags[tag].sort();
+  };
+
+  // Returns all the tags ("kidFriendly" and "adult" so far) in alphabetical order
+  App.allTags = function() {
+    return Object.keys(tags).sort();
+  };
+
+  // @param isIndoor: true (is indoor) or false (is outdoor)
+  App.indoorActivities = function(isIndoor) {
+    return listOfActivities.filter(function(activity) {
```

Comment:
You can get sorting "for free" if you refactor this to use `App.allActivities` instead of directly accessing the data structure `listOfActivities`.

That'd be DRY'ing up the code, with an immediate benefit.